### ISSUE TITLE
Provide feedback when interacting with PR files

### ISSUE THREAD

**Issue by danielo515** [microsoft/vscode-pull-request-github#3201]
Issue Type: <b>Feature Request</b>

Hello.
This extensionis super nice, it is super useful to be able to interact with PR direcrtly from VSCode.

There is one small detail that you may not notice if you are always on a good internet connection, but it becomes noticeable on slow ones. When you interact with any file on the PR, for example you flag it as viewed, there is no feedback to tell you if something is loading, or even if you clicked the button. It will be very nice to have some visual indication.

Regards

Extension version: 0.34.1
VS Code version: Code 1.63.2 (899d46d82c4c95423fb7e10e68eba52050e30ba3, 2021-12-15T09:37:28.172Z)
OS version: Darwin x64 20.6.0
Restricted Mode: No

**Comment by alexr00** [comment]
@danielo515 is this specific to marking files as viewed? Related: https://github.com/microsoft/vscode-pull-request-github/issues/3191

**Comment by danielo515** [comment]
Yes, it is related. But that issue is just about it being slow, this is about providing some UI "while waiting"

**Comment by ek-work** [comment]
A nice feature would be to be able to mark a `folder` as viewed instead of having to mark all files individually. The example I have is a huge PR with hundreds of files removed in a folder that doesn't actually need individual review. It's painful having to mark each one individually.

**Comment by alexr00** [comment]
> A nice feature would be to be able to mark a folder as viewed instead of having to mark all files individually. The example I have is a huge PR with hundreds of files removed in a folder that doesn't actually need individual review. It's painful having to mark each one individually.@ek-work you can a

**Comment by alexr00** [comment]
@ek-work if you check a folder level checkbox then it marks all the children as viewed as well.

**Comment by ek-work** [comment]
When I am in PR view I see what is shown in the screenshot below (this is from the docs - https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/filtering-files-in-a-pull-request):
![image](https://github.com/microsoft/vscode-pull-request-github/assets/7796057/05729f0f-d4af-43c0-b279-4e3fcfca3a0d)
I see the tree structrure on the left and I see a `Viewed` box for checking each individual file, but I seen no `folder level checkbox`. Is there some other view where that is available?

**Comment by alexr00** [comment]
@ek-work this repository is for the VS Code extension "GitHub Pull Requests and Issues". The checkboxes are in that extension:

![image](https://github.com/microsoft/vscode-pull-request-github/assets/38270282/0385362c-666b-4100-8a44-2791fc115bc0)
